pyogrio/_ogr.pxd

```diff
@@ -191,11 +191,20 @@ cdef extern from "ogr_srs_api.h":
 
 
 cdef extern from "arrow_bridge.h":
+    struct ArrowArray:
+        void (*release)(struct ArrowArray*)
```

Comment:
```suggestion
        void (*release)(ArrowArray*)
```

If you want to see a working version, see eg https://github.com/apache/arrow/blob/c67d0260d4e96472b5cbdff66ca67ead2b9abe4c/python/pyarrow/includes/libarrow.pxd#L2795-L2803

Cython generally leaves out "struct" in signatures

I think this causes some of the compilation errors

Reply:
Or nanoarrow-python has the full declaration, but it's not committed to github but autogenerated, so you can only see it locally. Copying what I have:

```
    struct ArrowSchema:
        const char* format
        const char* name
        const char* metadata
        int64_t flags
        int64_t n_children
        ArrowSchema** children
        ArrowSchema* dictionary
        void (*release)(ArrowSchema*)
        void* private_data

    struct ArrowArray:
        int64_t length
        int64_t null_count
        int64_t offset
        int64_t n_buffers
        int64_t n_children
        const void** buffers
        ArrowArray** children
        ArrowArray* dictionary
        void (*release)(ArrowArray*)
        void* private_data

    struct ArrowArrayStream:
        int (*get_schema)(ArrowArrayStream*, ArrowSchema* out)
        int (*get_next)(ArrowArrayStream*, ArrowArray* out)
        const char* (*get_last_error)(ArrowArrayStream*)
        void (*release)(ArrowArrayStream*)
        void* private_data
``

Reply:
Updated in https://github.com/geopandas/pyogrio/pull/346/commits/331599852e1b0ac8a9bddd686fa834f24d604939